omod/src/main/webapp/resources/scripts/custom/deletePatient.js

```diff
@@ -0,0 +1,46 @@
+var delPatient = delPatient || {};
+
+delPatient.deletePatientCreationDialog = null;
+
+/**
+ * Functions used to Delete a Patient (uses DeletePatientFragmentController)
+ */
+
+visit.returnUrl = "/coreapps/findpatient/findPatient.page?app=coreapps.findPatient";
+
+delPatient.createDeletePatientCreationDialog = function() {
+    delPatient.deletePatientCreationDialog = emr.setupConfirmationDialog({
+        selector: '#delete-patient-creation-dialog',
+        actions: {
+            confirm: function() {
+                emr.getFragmentActionWithCallback('coreapps', 'deletePatient', 'deletePatientEntry',
+                    { patient: delPatient.patient},
+                    function(data) {
+                        emr.successMessage(data.message);
+                        jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
+                        delPatient.deletePatientCreationDialog.close();
+                        delPatient.goToReturnUrl();
+                    }, function(err) {
+                        alert("Unable to delete user");
```

Comment:
This is where it is being redirected to. As posted in the ticket, error stack is:

> ERROR - JDBCExceptionReporter.logExceptions(234) |2016-04-15 02:51:10,698| Cannot delete or update a parent row: a foreign key constraint fails (`openmrs`.`encounter`, CONSTRAINT `en
> counter_patient` FOREIGN KEY (`patient_id`) REFERENCES `patient` (`patient_id`) ON UPDATE CASCADE)
> ERROR - LoggingAdvice.invoke(135) |2016-04-15 02:51:10,774| An error occurred while executing this method.
> Current user: admin
> Error message: could not delete: [org.openmrs.Patient#7]; SQL [delete from patient where patient_id=?]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViola
> tionException: could not delete: [org.openmrs.Patient#7]
> org.springframework.dao.DataIntegrityViolationException: could not delete: [org.openmrs.Patient#7]; SQL [delete from patient where patient_id=?]; constraint [null]; nested exception
> is org.hibernate.exception.ConstraintViolationException: could not delete: [org.openmrs.Patient#7]